client/extensions/woocommerce/components/reading-widget/index.js

```diff
@@ -0,0 +1,197 @@
+
+		return (
+			<div>
+				<QueryPosts
+					query={ query }
+					siteId={ siteId }
+				/>
+				<ul className="reading-widget__article-list">
+					{
+						( posts && posts.map( post => {
+							const { date, ID, title, URL } = post;
+							return (
+								<li key={ ID }>
+									<ExternalLink icon={ true } href={ URL } rel="noopener noreferrer" target="_blank">
+										{ title }
+									</ExternalLink>
+									<span>{ humanDate( date ) }</span>
+								</li>
+							);
+						} ) )
+					}
+				</ul>
+			</div>
+		);
+	}
+
+	renderSubscriptionFormExplanation = ( expanded ) => {
+		const { translate } = this.props;
+
+		return (
+			<div className="reading-widget__subscription-form-explanation">
+				<p>
+					{ translate( 'Subscribe to our newsletter and get the latest' +
+						' news delivered to your inbox' ) }
+				</p>
+				{
+					expanded && ( <p>
+						{ translate( 'Before you subscribe, please specify your role(s)' +
+							' in eCommerce and we\'ll send you the most relevant news & info' ) }
+					</p> )
+				}
+			</div>
+		);
+	}
+
+	onEmailFocus = () => {
+		this.setState( {
+			subscriptionFormExpanded: true,
+		} );
+	}
+
+	onEmailChange = ( e ) => {
+		this.setState( {
+			emailAddress: e.target.value
+		} );
+	}
+
+	onRolesChange = ( e ) => {
+		this.setState( {
+			roles: e.value
+		} );
+	}
+
+	isFormSubmittable = () => {
+		const emailValidates = emailValidator.validate( this.state.emailAddress );
+		const roleSelected = ( 0 < this.state.roles.length );
+
+		return emailValidates && roleSelected;
+	}
+
+	onSubmit = () => {
+		// TODO
+	}
+
+	renderSubscriptionFormFields = ( expanded ) => {
+		const { translate } = this.props;
+		return (
+			<div className="reading-widget__subscription-form-fields">
+				<FormTextInput
+					onChange={ this.onEmailChange }
+					onFocus={ this.onEmailFocus }
+					placeholder={ translate( 'hello@email.com' ) }
+					value={ this.state.emailAddress }
+				/>
+				{ expanded && (
+					<MultiCheckbox
+						checked={ this.state.roles }
+						onChange={ this.onRolesChange }
+						options={ this.getRolesAndLabels() }
+					/>
+				) }
+				{ expanded && (
+					<Button
+						disabled={ ! this.isFormSubmittable() }
+					>
+						{ translate( 'Subscribe' ) }
+					</Button>
+				) }
+			</div>
+		);
+	}
+
+	renderSubscriptionForm = () => {
+		const expanded = this.state.subscriptionFormExpanded;
+
+		return (
+			<div className="reading-widget__subscription-form">
+				{ this.renderSubscriptionFormExplanation( expanded ) }
+				{ this.renderSubscriptionFormFields( expanded ) }
+			</div>
+		);
+	}
+
+	render = () => {
+		return (
+			<div className="reading-widget__container card">
+				{ this.renderHeading() }
+				{ this.renderArticleList() }
+				{ this.renderSubscriptionForm() }
+			</div>
+		);
+	}
+}
+
+export default connect( ( state, ownProps ) => {
+	const siteId = ownProps.siteId ? ownProps.siteId : ReadingWidget.defaultProps.siteId;
```

Comment:
Not yet, but since we will probably want to be able to set a different query for "momma" than fledgling (e.g. a different category) it made sense to me to allow setting the blog ID as well.